examples/nlp/token_classification/conf/punctuation_capitalization_config.yaml

```diff
@@ -39,50 +41,84 @@ exp_manager:
   create_checkpoint_callback: true  # Whether you want exp_manager to create a model checkpoint callback
 
 model:
-  punct_label_ids: null # will be filled during training
-  capit_label_ids: null # will be filled during training
-  class_labels:
-    punct_labels_file: punct_label_ids.csv
-    capit_labels_file: capit_label_ids.csv
-  dataset:
-    data_dir: ??? # /path/to/training/data
-    # parameters for dataset preprocessing
-    max_seq_length: 128
+  common_dataset_parameters:
     pad_label: 'O'
     ignore_extra_tokens: false
     ignore_start_end: false
-    use_cache: true
-    # shared among dataloaders
-    num_workers:  2
-    pin_memory: false
-    drop_last: false
+    punct_label_ids: null
+    punct_label_vocab_file: null
+    capit_label_ids: null
+    capit_label_vocab_file: null
 
   train_ds:
+    # Tarred dataset is recommended if all dataset cannot be loaded in memory. Use script
+    # `examples/nlp/token_classification/create_punctuation_capitalization_tarred_dataset.py` for tarred dataset
+    # creation.
+    use_tarred_dataset: false
+    # A path to directory where `tar_metadata_file` is stored. If `ds_item` is missing, then `model.dataset.data_dir`
+    # will be used instead. You cannot pass list of paths in `model.train_ds.ds_item`.
+    ds_item: ???
+
     text_file: text_train.txt
     labels_file: labels_train.txt
+    # Permutes batches every epoch
     shuffle: true
     num_samples: -1
-    batch_size: 64
+    tokens_in_batch: 15000
+    # Number of jobs for tokenization and labels encoding. If 0, then multiprocessing is not used. If null,
+    # number of jobs is equal to the number of CPU cores.
+    # WARNING: can cause deadlocks with tokenizers, which use multiprocessing (e.g. SentencePiece)
+    n_jobs: 0
+
+    # Path to tarred dataset metadata file. Required if tarred dataset is used. Metadata file is a JSON file which
+    # contains total number of batches in the dataset, a list of paths to tar files and paths to label vocabularies.
+    # Metadata file is create by script
+    # `examples/nlp/token_classification/create_punctuation_capitalization_tarred_dataset.py`
+    tar_metadata_file: null
+    # Controls batch shuffling in tarred dataset. `tar_shuffle_n` is a size of shuffled batch buffer. Mind that this
+    # shuffling only permutes batches and doesn't exchange samples between batches. Proper shuffling is turned on in
+    # usual dataset.
+    tar_shuffle_n: 1
```

Comment:
How about moving tar_shuffle_n, tart_metadata_file, and n_jobs right after use_tarred - with the note that these are only for tarred dataset, tokens_n_batch - for both tar and regular dataset, the rest args for regular format.